x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/20_search_application_put.yml

```diff
@@ -216,14 +216,7 @@ teardown:
       search_application.get:
         name: test-updated-search-application
 
-  - match:
-      template:
-        script:
-          source: "{\n  \"query\": {\n    \"query_string\": {\n        \"query\": \"{{query_string}}\",\n        \"default_field\": \"{{default_field}}\"\n        }\n    }\n}\n"
-          lang: "mustache"
-          params:
-            query_string: "*"
-            default_field: "*"
+  - match: { template: null }
```

Comment:
Shouldn't the default template be retrieved on a GET? I think users need to understand what would be the  template applied.

It's fine not to store it, but for example if we change the default template for future releases, users should be able to retrieve it.

WDYT?

Reply:
Second thoughts 🤔  - maybe users want to understand whether this is the default template or not. 

I think it would be more informative to return it anyway - maybe adding a `default` field in the `template` to help users understand that?

Reply:
Thank you for this question and discussion, I thought a bit on this as well and can see arguments for both cases. 

Why I decided to go with null: 

- It accurately reflects the state of what's stored (you are using the default template) 
- It may not be quickly apparent - even with a default flag - that they're using the default field and they didn't just store this field 
- We document what the default template is and what is set when a template is not specified. 
- The docs also point to the `_application/search_application/render_metadata` field to allow "test runs" of the query that's generated with parameters they're sending in, so they could tune this a bit more. 

I believe that's sufficient and clear, but I'm happy to discuss more.

Reply:
> I think it would be more informative to return it anyway - maybe adding a default field in the template to help users understand that?

💯 on this - the default template might change between versions - if we do not return it, we force users to check the documentation to find out what the default is. if we want to build a UI in Kibana to manage/tweak the search application template, we would also have to keep an evidence of what the default template is for each version.

Reply:
Per discussion offline, we'll return the default template, but instead of a `default` field add a warning header. Thanks all for the discussion!